proto/ibc/applications/interchain_accounts/v1/metadata.proto

```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+
+package ibc.applications.interchain_accounts.v1;
+
+option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/types";
+
+import "gogoproto/gogo.proto";
+
+// Metadata defines a set of protocol specific data encoded into the ICS27 channel version bytestring
+// See ICS004: https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#Versioning
+message Metadata {
+  string version                  = 1;
+  string controller_connection_id = 2 [(gogoproto.moretags) = "yaml:\"controller_connection_id\""];
+  string host_connection_id       = 3 [(gogoproto.moretags) = "yaml:\"host_connection_id\""];
+  string address                  = 4;
```

Comment:
Should we be a bit more specific and call this something like `ica_address`?

Reply:
I'm not sure, I think its probably fine but if anyone feels strongly about it we can change

Reply:
probs better to be explicit

Reply:
I'll handle this in a follow up with the other changes mentioned [above](https://github.com/cosmos/ibc-go/pull/700#discussion_r781519317)